### PR TITLE
horizon: hide is_authorized flag for native asset

### PR DESCRIPTION
**Pull request by howardtw**
We don't want to show the flag for XLM.

### CODE REVIEW

**Comment by tomquisel** [comment]
@howardtw looks good! Can you throw in a test that verifies that the Horizon API response doesn't come back with `is_authorized` for the native asset?

**Comment by howardtw** [comment]
@tomquisel does [this](https://github.com/stellar/go/pull/1014/commits/a30f5b7595cbb41625cc1d2c5ec20513ebcc0f79) work? Or you want to see a test parsing the response.body and check whether the string `is_authorized` exists?

**Comment by howardtw** [comment]
It will be a similar fashion that we declare a new struct like the following:
```
type auth struct {
  IsAuthorized *bool `json:is_authorzied`
}
```
We will still be checking whether the field is `nil`

File: services/horizon/internal/resourceadapter/balance_test.go [hunk 1]
```diff
@@ -49,7 +50,8 @@ func TestPopulateBalance(t *testing.T) {
 	assert.Equal(t, "0.0000100", want.Limit)
 	assert.Equal(t, "", want.Issuer)
 	assert.Equal(t, testAssetCode2, want.Code)
-	assert.Equal(t, false, want.IsAuthorized)
+	vFalse := false
+	assert.Equal(t, &vFalse, want.IsAuthorized)
```
Comment (bartekn) [inline services/horizon/internal/resourceadapter/balance_test.go]: nit: could be `assert.Equal(t, false, *want.IsAuthorized)`

Comment (howardtw) [reply services/horizon/internal/resourceadapter/balance_test.go]: I like it much better.

**Pull request merged**